packages/expo-modules-core/ios/Core/AppContext.swift

```diff
@@ -406,6 +423,8 @@ public final class AppContext: NSObject {
     let runtime = try runtime
     let coreObject = runtime.createObject()
 
+    coreObject.defineProperty("__expo_appIdentifier", value: appIdentifier == 0 ? "" : "\(appIdentifier)", options: [])
```

Comment:
I would say it's better for the identifier to be `null` when it's not available rather than an empty string.
Also, for the shared object IDs we're using double underscores at the end (`__expo_shared_object_id__`), so we may want to be consistent with that.

```suggestion
    coreObject.defineProperty("__expo_appIdentifier__", value: appIdentifier == 0 ? nil : "\(appIdentifier)", options: [])
```